File: app/src/main/java/workshop/akbolatss/tagsnews/model/RssItemRepository.java
package workshop.akbolatss.tagsnews.model;

import java.util.List;

import io.reactivex.Observable;
import me.toptas.rssconverter.RssItem;

public interface RssItemRepository {

    void addRssItem(RssItem rssItem);

    void removeRssItem(String pubDate);

    boolean checkIfExists(String pubDate);

    Observable<List<RssItem>> getFavorites();
}

File: app/src/main/java/workshop/akbolatss/tagsnews/model/RssItemRepository.kt
package workshop.akbolatss.tagsnews.model

import io.reactivex.Observable
import workshop.akbolatss.tagsnews.model.dao.RssFeedItem

/**
 * @see DBRssItemRepository
 */
interface RssItemRepository {

    val favorites: Observable<List<RssFeedItem>>

    fun addRssItem(rssFeedItem: RssFeedItem)

    fun removeRssItem(rssFeedItem: RssFeedItem)

    fun checkIfExists(rssFeedItem: RssFeedItem): Boolean
}
